pkg/fwruleprocessor/fw_rule_processor.go

```diff
@@ -212,16 +193,38 @@ func addCatchAllL4Entry(firewallRule *EbpfFirewallRules) {
 	firewallRule.L4Info = append(firewallRule.L4Info, catchAllL4Entry)
 }
 
+func addDenyAllL4Entry(firewallRule *EbpfFirewallRules) {
+	denyAllL4Entry := v1alpha1.Port{
+		Protocol: &DENY_ALL_PROTOCOL,
+	}
+	firewallRule.L4Info = append(firewallRule.L4Info, denyAllL4Entry)
+}
+
 func checkAndDeriveL4InfoFromAnyMatchingCIDRs(firewallRule string,
-	nonHostCIDRs map[string][]v1alpha1.Port) []v1alpha1.Port {
+	cidrsMap map[string]EbpfFirewallRules) []v1alpha1.Port {
 	var matchingCIDRL4Info []v1alpha1.Port
 
 	_, ipToCheck, _ := net.ParseCIDR(firewallRule)
-	for nonHostCIDR, l4Info := range nonHostCIDRs {
-		_, cidrEntry, _ := net.ParseCIDR(nonHostCIDR)
+	for cidr, cidrFirewallInfo := range cidrsMap {
+		if !utils.IsNonHostCIDR(cidr) {
+			continue
+		}
+		_, cidrEntry, _ := net.ParseCIDR(cidr)
```

Comment:
Do we need check parsing error to avoid / catch unexpected failures in following logics?